stetho/src/main/java/com/facebook/stetho/inspector/network/ResponseBodyFileManager.java

```diff
@@ -83,4 +122,48 @@ public OutputStream openResponseBodyFile(String requestId, boolean base64Encode)
   private static String getFilename(String requestId) {
     return FILENAME_PREFIX + requestId;
   }
+
+  /**
+   * Associates an asynchronous pretty printer with a response request id
+   * The pretty printer will be used to pretty print the response body that has
+   * the particular request id
+   *
+   * @param requestId Unique identifier for the response
+   * as per {@link NetworkEventReporter.InspectorResponse#requestId()}
+   * @param asyncPrettyPrinter Asynchronous Pretty Printer to pretty print the response body
+   */
+  public void associateAsyncPrettyPrinterWithId(
+      String requestId,
+      AsyncPrettyPrinter asyncPrettyPrinter) {
+    if (mRequestIdMap.containsKey(requestId)) {
+      throw new IllegalArgumentException("cannot associate different " +
+          "pretty printers with the same request id: "+requestId);
+    }
+    mRequestIdMap.put(requestId, asyncPrettyPrinter);
+  }
+
+  private class AsyncPrettyPrintingTask implements Callable<String> {
```

Comment:
`Task` is an overloaded term on Android and usually implies an `AsyncTask`.  Since you don't have this here, instead call this `Job` or `Callable`.